rust/noosphere-gateway/src/gateway.rs

```diff
@@ -79,64 +65,64 @@ where
+    let (name_system_tx, name_system_task) = start_name_system::<M, C, S>(
         NameSystemConfiguration {
             connection_type: NameSystemConnectionType::Remote(name_resolver_api),
             ipfs_api,
         },
-        vec![sphere_context.clone()],
+        context_manager.clone(),
     );
-    let (cleanup_tx, cleanup_task) = start_cleanup::<C, S>(sphere_context.clone());
+    let (cleanup_tx, cleanup_task) = start_cleanup::<M, C, S>(context_manager.clone());
 
     let app = Router::new()
         .route(
             &v0alpha1::Route::Did.to_string(),
-            get(handlers::v0alpha1::did_route),
+            get(handlers::v0alpha1::did_route::<M, C, S>),
         )
         .route(
             &v0alpha1::Route::Replicate(None).to_string(),
-            get(handlers::v0alpha1::replicate_route::<C, S>),
+            get(handlers::v0alpha1::replicate_route::<M, C, S>),
         )
         .route(
             &v0alpha1::Route::Identify.to_string(),
-            get(handlers::v0alpha1::identify_route::<C, S>),
+            get(handlers::v0alpha1::identify_route::<M, C, S>),
         )
         .route(
             &v0alpha1::Route::Push.to_string(),
             #[allow(deprecated)]
-            put(handlers::v0alpha1::push_route::<C, S>),
+            put(handlers::v0alpha1::push_route::<M, C, S>),
         )
         .route(
             &v0alpha2::Route::Push.to_string(),
-            put(handlers::v0alpha2::push_route::<C, S>),
+            put(handlers::v0alpha2::push_route::<M, C, S>),
         )
         .route(
             &v0alpha1::Route::Fetch.to_string(),
-            get(handlers::v0alpha1::fetch_route::<C, S>),
+            get(handlers::v0alpha1::fetch_route::<M, C, S>),
         )
-        .layer(Extension(sphere_context.clone()))
-        .layer(Extension(gateway_scope.clone()))
+        .layer(Extension(context_manager))
         .layer(Extension(ipfs_client))
-        .layer(Extension(gateway_key_did))
         .layer(Extension(syndication_tx))
         .layer(Extension(name_system_tx))
         .layer(Extension(cleanup_tx))
         .layer(DefaultBodyLimit::max(DEFAULT_BODY_LENGTH_LIMIT))
         .layer(cors)
         .layer(TraceLayer::new_for_http());
 
-    info!(
-        r#"A geist is summoned to manage local sphere {}
-
-It has bound a gateway to {:?}
-It awaits updates from sphere {}..."#,
-        gateway_scope.identity,
-        listener
-            .local_addr()
-            .expect("Unexpected missing listener address"),
-        gateway_scope.counterpart
-    );
+    /*
+        info!(
+            r#"A geist is summoned to manage local sphere {}
+
+    It has bound a gateway to {:?}
+    It awaits updates from sphere {}..."#,
+            gateway_scope.identity,
+            listener
+                .local_addr()
+                .expect("Unexpected missing listener address"),
+            gateway_scope.counterpart
+        );
+        */
```

Comment:
Possibly:
* ContextManager calls a success callback on successful server start that can print out a message
* Return a task or wrapper from `start_gateway()` that needs to be awaited to actually start the server, giving consumer a moment to log whatever

Neither feels correct compared to exposing routes and a (new) worker abstraction, such that the STG is the only caller of `start_gateway` (and we can then leave it in), but maybe a ContextManager callback is the quickest path forward

Reply:
I think what we need to do is extract all the Axum setup stuff, so that you can get an `app` without any superfluous logging and then invoke `Server::from_tcp(listener)?.serve(app.into_make_service())` separately.


Then, we can have two different `start_gateway`-like  entrypoints: one that may be implemented for the CLI, and one that may be implemented for cloud use.

Reply:
Implemented via

```rust
let gateway = Gateway::new(manager, ipfs_url, ns_url, None)?;
// log here
gateway.start(listener).await?;
```